### ISSUE TITLE
Providing an invalid webhook URL to `predictions.create` doesn't return any feedback

### ISSUE THREAD

**Issue by carlpeaslee** [replicate/replicate-javascript#81]
How to reproduce: 

```typescript
  await replicate.predictions.create({
    version: "6282abe6a492de4145d7bb601023762212f9ddbbe78278bd6771c8b3b2f2a13b", //vicuna
    input: {
      prompt: "this is definitely developer error  but it would be a nice quality of life improvement to surface the mistake",
    },
    webhook: `htp\\someinvalid,url.corn/webhawks`,
    webhook_events_filter: ['completed', 'logs', 'output', 'start'],
  })
```

The prediction will successfully run and can be verified in one's dashboard – it just never surfaces any errors or notifications that it couldn't deliver the webhook event.

I think in an ideal world `predictions.create` would probably:
- error when provided a structurally invalid URL
- still function if given a structurally valid but possibly non-responsive endpoint –– but then surface an error/warning somewhere for the user (either at an errors event endpoint or in the dashboard)

Not urgent just a nice to have that I ran in to today :)

**Comment by mattt** [comment]
Thanks for sharing this, @carlpeaslee. I think it's totally reasonable for the client library to help guard against this.

What do you think about updating the type definition for webhook parameters to explicitly look for things matching a URL pattern? I don't want to, like, enforce passing `URL` objects, and attempting to validate inputs at runtime feels wrong (that's more the API's job), but letting TypeScript warn at at build time feels helpful and unobtrusive.

**Comment by carlpeaslee** [comment]
So I'll give you the actual real example that I experienced: 

<img width="710" alt="image" src="https://github.com/replicate/replicate-javascript/assets/15232222/b3d534e8-5338-404c-84e5-b29f1ea74544">

Which I don't think would get caught by a typescript definition. 

I think you're right that this is probably more the API's job, I just couldn't find a repo for your API to leave an issue on :)

**Comment by mattt** [comment]
@carlpeaslee Ah, thanks for providing that example. In that case, a runtime check using the built-in `URL` constructor makes sense.

> I think you're right that this is probably more the API's job, I just couldn't find a repo for your API to leave an issue on :)

Oh yeah — sorry to imply that this wasn't the right place to file an issue. Our API isn't open source, so this is as good a place as any. 😄

**Comment by mattt** [comment]
@carlpeaslee Thanks again for your help with this! I just merged #84, which implements what we discussed. If you update to [v0.12.3](https://github.com/replicate/replicate-javascript/releases/tag/v0.12.3), you'll get an error the next time you pass an invalid URL.
